### PR TITLE
[FSDE-17447] Created Checkbox web component

### PR DESCRIPTION
**Pull request by janibolkvadze**

### CODE REVIEW

File: test/chi/custom-elements/checkbox.pug [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Checkbox
+---
+script(type='module' src='../../js/ce/ux-chi-ce/ux-chi-ce.esm.js')
+script(nomodule='' src='../../js/ce/ux-chi-ce/ux-chi-ce.js')
+
+h3 Base
```
Comment (janibolkvadze) [inline test/chi/custom-elements/checkbox.pug:7]: I left these to be used in the next iteration for adding documentation/tests.

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-360/1/. :x:

File: src/custom-elements/src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  _emitChange(checked: boolean) {
+    this.chiChange.emit(checked);
+  }
+
+  _updateIndeterminate() {
+    if (this.indeterminate) {
+      this.input.indeterminate = true;
+    } else {
+      this.input.indeterminate = false;
+    }
```
[the remark this answers is not in the document]
Comment (janibolkvadze) [reply src/custom-elements/src/components/checkbox/checkbox.tsx:66]: Makes sense, thanks!

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c41dec2fb1b436264c450ac86d2cac95f4a443cd-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-360-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Review by SergioQCL — APPROVED** [review]
BMB!

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-360/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c504e8c727d076ce415028d1f3e28ae594aedd95-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-360-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Review by jllr — APPROVED** [review]

**Pull request merged**